esx_service/utils/auth_api.py

```diff
@@ -365,11 +365,72 @@ def is_vm_duplicate(vm_list):
 
     return None
 
+def check_default_datastore(datastore_name):
+    """ Check datastore with given name is a valid value for default_datastore """
+    # The valid default_datastore name are:
+    # named datastore existing on the host
+    # hard coded datastore name "__VM_DS"
+    # "__ALL_DS" is not a valid value to set as "default_datastore"
+    if datastore_name == auth_data_const.VM_DS:
+        return None
+    if datastore_name == auth_data_const.ALL_DS:
+        return error_code.generate_error_info(ErrorCode.DS_DEFAULT_NAME_INVALID, datastore_name)
```

Comment:
FWIW, the length of line making string out of code is really annoying.
As a simple and partial fix, can you change 'from ErrCode import Error code' to "from ErrorCode import ErrorCode, generate_code_error, error_code_to_message' (or simply 'from ErrorCode import *' and drop error_code. prefix everywhere ?